api/types/filters/parse.go

```diff
@@ -196,6 +196,37 @@ func (args Args) Match(field, source string) bool {
 	return false
 }
 
+// GetBoolOrDefault returns a boolean value of the key if the key is present
+// and is intepretable as a boolean value. Otherwise the default value is returned.
+// Error is not nil only if the filter values are not valid boolean or are conflicting.
+func (args Args) GetBoolOrDefault(key string, defaultValue bool) (bool, error) {
+	fieldValues, ok := args.fields[key]
+
+	if !ok {
+		return defaultValue, nil
+	}
+
+	if len(fieldValues) == 0 {
+		return defaultValue, invalidFilter{key, nil}
+	}
+
+	isFalse := fieldValues["0"] || fieldValues["false"]
+	isTrue := fieldValues["1"] || fieldValues["true"]
+
+	conflicting := isFalse && isTrue
+	invalid := !isFalse && !isTrue
+
+	if conflicting || invalid {
+		return defaultValue, invalidFilter{key, args.Get(key)}
+	} else if isFalse {
+		return false, nil
+	} else if isTrue {
+		return true, nil
+	}
+
+	return defaultValue, fmt.Errorf("unexpected code path reached for values=%s", args.Get(key))
```

Comment:
I wanted to avoid importing errdefs (it's not imported in this file, the `invalidFilter` just implements the `InvalidParameter` function, and thought that because this shouldn't happen anyway, then the "FIXME" message would at least be visible in the logs if it somehow ended up taking this code path 😄 
I can add a helper struct that implements the `errdefs.ErrSystem` if you feel it would be better though.

Reply:
Ah, right, yes. So ideally implementing a type for it, although it looks like `errdefs` is already imported for other parts, so probably not a huge deal to use it.

I'm fine with looking at that separately though; not a blocker.

Reply:
Ah, it seems that errdefs somehow made it's way to the `parse_test.go`. Looking at the change that caused is makes me think this was accidental.
I dropped the errdefs dependency completely and changed the error to error-type.